File: app/src/main/java/com/maple/googlemap/base/BaseFragment.java
package com.maple.googlemap.base;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;


/**
 * @author maple
 * @time 2018/12/17
 */
public abstract class BaseFragment extends Fragment {
    public View view;
    public Context mContext;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.mContext = getActivity();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(getLayoutRes(), container, false);
        view.setClickable(true);
        return view;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initData(savedInstanceState);
    }

    public boolean onKeyBackPressed() {
        // 是否消耗掉back事件
        return false;
    }

    public abstract int getLayoutRes();

    public abstract void initData(Bundle savedInstanceState);

}

File: app/src/main/java/com/maple/googlemap/base/BaseFragment.kt
package com.maple.googlemap.base

import android.content.Context
import androidx.fragment.app.Fragment

/**
 * @author maple
 * @time 2018/10/11
 */
abstract class BaseFragment : Fragment() {
    lateinit var mContext: Context
//    val mProgressDialog: ProgressDialog by lazy {
//        LoadingDialog().getDefault(context)
//    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mContext = context
    }

    open fun onKeyBackPressed(): Boolean {
        // 是否消耗掉back事件
        return false
    }

//    open fun showProgressDialog(@StringRes resString: Int) {
//        showProgressDialog(getString(resString))
//    }
//
//    open fun showProgressDialog(message: CharSequence) {
//        dismissDialog()
//        mProgressDialog.setMessage(message)
//        mProgressDialog.show()
//    }
//
//    open fun dismissDialog() {
//        if (mProgressDialog.isShowing) {
//            mProgressDialog.cancel()
//        }
//    }

}

